### PR TITLE
543001 - infinite scroll load function

### PR DESCRIPTION
**Pull request by JonathanToddMF**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue

https://internal.almoctane.com/ui/entity-navigation?p=131002/7002&entityType=work_item&id=543001

#### Description of Proposed Changes

reset loading state when resetting to allow multiple requests

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_543001-infiniteScroll-load-function

#### Downstream Build(s)

* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~543001-infiniteScroll-load-function~CI/)

### CODE REVIEW

File: src/directives/infinite-scroll/infinite-scroll.component.spec.ts [hunk 1]
```diff
@@ -133,3 +133,67 @@ describe('Directive - Infinite Scroll', () => {
...
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(InfiniteScrollTestDelayComponent);
+        component = fixture.componentInstance;
+        loadSpy = spyOn(component, 'load').and.callThrough();
+        fixture.detectChanges();
+    });
+
+    // checking if promise resolves twice
+    it('should resolve the load function when reset is called during load', () => {
```
Comment (donna-marijne) [inline src/directives/infinite-scroll/infinite-scroll.component.spec.ts:184]: It doesn't look like reset is called during load (or indeed at all).

File: src/directives/infinite-scroll/infinite-scroll.component.spec.ts [hunk 2]
```diff
@@ -133,3 +133,67 @@ describe('Directive - Infinite Scroll', () => {
...
+        expect(loadSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should allow loading to be triggered after reset, regardless if there are any pending requests', fakeAsync(() => {
+        component.infiniteScrollDirective.reset(true);
+        component.infiniteScrollDirective.check();
```
Comment (donna-marijne) [inline src/directives/infinite-scroll/infinite-scroll.component.spec.ts:190]: In the repro sandbox, they are only calling `reset()` without an argument and without calling `check`.
```
this.usersList.reset();
```
Could you add a test case which matches this scenario?

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Looks good. An e2e test might be more appropriate to this issue.

**Review by donna-marijne — APPROVED** [review]
Verification:
https://codesandbox.io/s/infinite-scroll-ux-aspects-micro-focus-forked-dtifcv?file=/package.json - :heavy_check_mark:

**Review by Roland-Penny — APPROVED** [review]
Looks good

**Pull request merged**